Simplenote/src/androidTest/java/com/automattic/simplenote/models/NoteIndexerTest.java

```diff
@@ -21,33 +21,80 @@
 @RunWith(AndroidJUnit4.class)
 @SmallTest
 public class NoteIndexerTest {
+    private static final String DATE_STRING = "2013-09-12T20:47:03.000Z";
+    private static final double DATE_DOUBLE = 1379018823037.0;
+    private static final long DATE_LONG = 1379018823000L;
+    private static final long DATE_NUMBER = 1379018823;
 
+    private Note mNote;
     private NoteIndexer mIndexer;
 
     @Before
-    public void setUp() {
+    public void setUp() throws JSONException {
+        JSONObject properties = new JSONObject();
+        properties.put(Note.CREATION_DATE_PROPERTY, DATE_DOUBLE);
+        properties.put(Note.MODIFICATION_DATE_PROPERTY, DATE_DOUBLE);
+        properties.put(Note.PUBLISH_URL_PROPERTY, "test.url.property");
+        mNote = new Note("test", properties);
         mIndexer = new NoteIndexer();
     }
 
     @Test
     public void testIndexMillisecondDates() throws JSONException {
-        JSONObject properties = new JSONObject();
-        properties.put(Note.MODIFICATION_DATE_PROPERTY, 1379018823037.0);
-        properties.put(Note.CREATION_DATE_PROPERTY, 1379018823037.0);
+        List<Index> indexes = mIndexer.index(mNote);
+        assertIndex(indexes, Note.MODIFIED_INDEX_NAME, DATE_LONG);
+        assertIndex(indexes, Note.CREATED_INDEX_NAME, DATE_LONG);
+    }
 
-        Note note = new Note("test", properties);
-        List<Index> indexes = mIndexer.index(note);
 
-        assertIndex(indexes, Note.MODIFIED_INDEX_NAME, 1379018823000L);
-        assertIndex(indexes, Note.CREATED_INDEX_NAME, 1379018823000L);
+    @Test
+    public void testConvertMillisToSeconds() {
+        Calendar date = Note.numberToDate(DATE_DOUBLE);
+        assertThat(date.getTimeInMillis(), is(DATE_LONG));
     }
 
+    @Test
+    public void testConvertNumberToDateString() {
+        String dateConverted = Note.numberToDateString(DATE_NUMBER);
+        assertThat(dateConverted, is(DATE_STRING));
+    }
 
     @Test
-    public void testConvertMillisToSeconds() {
-        Calendar date = Note.numberToDate(1379018823037.0);
+    public void testGetCreationDate() throws JSONException {
+        Calendar calendar = mNote.getCreationDate();
+        assertThat(calendar.getTimeInMillis(), is(DATE_LONG));
+    }
+
+    @Test
+    public void testGetCreationDateString() {
+        Calendar date = Note.numberToDate(DATE_DOUBLE);
+        mNote.setCreationDate(date);
+        assertThat(mNote.getCreationDateString(), is(DATE_STRING));
+    }
+
+    @Test
+    public void testGetModificationDate() throws JSONException {
```

Comment:
```suggestion
    public void testGetModificationDate() {
```

We can remove `throws JSONException` since it's unused.